### PR TITLE
[wasm] Fixes for issue #108519

### PR DESCRIPTION
**Pull request by kg**
Contributes to https://github.com/dotnet/runtime/issues/108519

### CODE REVIEW

**Comment by kg** [comment]
cc @pavelsavara

**Comment by pavelsavara** [comment]
this will need Net9 backport

File: src/mono/mono/mini/interp/jiterpreter.c [hunk 1]
```diff
@@ -1656,8 +1658,11 @@ get_queue (int queue) {
 // Purges this item from all queues
 void
 mono_jiterp_tlqueue_purge_all (gpointer item) {
-	mono_os_mutex_lock (&queue_mutex);
+	// HACK: Call get_queue_key to ensure the queues are initialized before enumerating them
```
Comment (pavelsavara) [inline src/mono/mono/mini/interp/jiterpreter.c:1661]: why not `if (!mono_jiterp_is_enabled()) return;` ?

**Review by pavelsavara — APPROVED** [review]

File: src/mono/mono/mini/interp/jiterpreter.c [hunk 2]
```diff
@@ -1645,8 +1646,9 @@ get_queue (int queue) {
 	GPtrArray *result = NULL;
 	if ((result = (GPtrArray *)mono_native_tls_get_value (key)) == NULL) {
 		g_assert (mono_native_tls_set_value (key, result = g_ptr_array_new ()));
-		mono_os_mutex_lock (&queue_mutex);
 		// WARNING: Ensure we do not call into the runtime or JS while holding this mutex!
```
Comment (dlemstra) [inline src/mono/mono/mini/interp/jiterpreter.c:1649]: Just a tiny style comment. In the rest of the code this comment is after the `mono_os_mutex_lock` call.

**Review by lewing — APPROVED** [review]

**Comment by dlemstra** [comment]
Thanks for the quick fix!

**Pull request merged**